lib/inky/inkycommands.ex

```diff
@@ -0,0 +1,192 @@
+defmodule Inky.Commands do
+  alias Inky.InkyIO
+
+  # API
+  def pixels_to_bitstring(pixels, display, p2b) do
+    rotation = display.rotation / 90
+    width = display.width
+    height = display.height
+
+    opts = bitstring_traversal_opts(rotation, width, height)
+    do_pixels_to_bitstring(pixels, p2b, opts)
+  end
+
+  def update(pins, display, packed_height, buffer_black, buffer_accent) do
+    pins
+    |> set_analog_block_control()
+    |> set_digital_block_control()
+    |> set_gate(packed_height)
+    |> set_gate_driving_voltage()
+    |> dummy_line_period()
+    |> set_gate_line_width()
+    |> set_data_entry_mode()
+    |> power_on()
+    |> vcom_register()
+    |> set_border_color()
+    |> configure_if_yellow(display.accent)
+    |> set_luts(display.luts)
+    |> set_dimensions(display.width, packed_height)
+    |> push_pixel_data(buffer_black, buffer_accent)
+    |> display_update_sequence()
+    |> trigger_display_update()
+    |> wait_before_sleep()
+    |> deep_sleep()
+  end
+
+  def soft_reset(pins) do
+    send_command(pins, 0x12)
+  end
+
+  def await_device(pins) do
+    should_wait = InkyIO.read_busy(pins) == 1
+
+    if should_wait do
+      :timer.sleep(10)
+      await_device(pins)
+    else
+      :ok
+    end
+  end
+
+  # Bitstringifying
+
+  defp bitstring_traversal_opts(rotation, width, height) do
+    case rotation do
+      -1.0 -> %{order: :x_outer, i_1: width - 1, i_n: 0, j_1: 0, j_n: height - 1}
+      1.0 -> %{order: :x_outer, i_1: 0, i_n: width - 1, j_1: height - 1, j_n: 0}
+      -2.0 -> %{order: :y_outer, i_1: width - 1, i_n: 0, j_1: height - 1, j_n: 0}
+      _ -> %{order: :y_outer, i_1: 0, i_n: height - 1, j_1: 0, j_n: width - 1}
+    end
+  end
+
+  defp do_pixels_to_bitstring(pixels, p2b, opts) do
+    for i <- opts.i_1..opts.i_n,
+        j <- opts.j_1..opts.j_n,
+        do: <<pixel_value(pixels, p2b, opts.order, i, j)::size(1)>>,
```

Comment:
Pulled out a few levels of the for-loop into the pixel_value function since they tried to enumerate a tuple and an integer which did not work.

Reply:
D'oh. Nice!